File: test2/src/main/java/xyz/yggdrazil/delaunay/voronoi/nodename/as3delaunay/Triangle.java
package xyz.yggdrazil.delaunay.voronoi.nodename.as3delaunay;

import java.util.ArrayList;

public final class Triangle {

    private ArrayList<Site> sites;

    public Triangle(Site a, Site b, Site c) {
        sites = new ArrayList();
        sites.add(a);
        sites.add(b);
        sites.add(c);
    }

    public ArrayList<Site> getSites() {
        return sites;
    }

    public void dispose() {
        sites.clear();
        sites = null;
    }
}

File: test2/src/main/kotlin/xyz/yggdrazil/fortune/Triangle.kt
package xyz.yggdrazil.fortune

import java.util.ArrayList

class Triangle(a: Site, b: Site, c: Site) {

    var sites: ArrayList<Site>? = null
        private set

    init {
        sites = ArrayList()
        sites!!.add(a)
        sites!!.add(b)
        sites!!.add(c)
    }

    fun dispose() {
        sites!!.clear()
        sites = null
    }
}